SecurityPkg/Test/Mock/Include/GoogleTest/Ppi/MockTcgPpi.h

```diff
@@ -0,0 +1,122 @@
+/** @file MockTcgPpiLib.h
+  Google Test mocks for TcgPpi
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#ifndef MOCK_TCG_PPI_H__
+
+#define __MOCK_TCG_PPI_H__
+
+#include <Library/GoogleTestLib.h>
+#include <Library/FunctionMockLib.h>
+
+extern "C" {
+  #include <PiPei.h>
+
+  // #include <Ppi/Tcg.h>
+  // NOTE: Directly includes Ppi/Tcg.h will cause a compilation error when building
+  // google test driver. Because of in MdePkg/Include/IndustryStandard/Tpm12.h,
+  // there is a C++ keyword `operator` is used in the TPM_PERMANENT_FLAGS structure.
+  // To workaround this I moved all necessary definitions for TcgPpi into this file.

```

Comment:
I had worked on a fix to Tpm12.h to address this c++ keyword issue.  Must have been dropped.  I need to repost

Reply:
See #6461

Reply:
Once #6461 completed, I will update `MockTcgPpi.h` in this PR to remove the redundant contents.